### PR TITLE
Docs: updated product names

### PR DESCRIPTION
**Pull request by goransle**
Updated product names with `Highcharts Stock` and `Highcharts Maps` in place of `Highstock`and `Highmaps`. Also added links for Gantt to the index page.

### CODE REVIEW

File: docs/Readme.md [hunk 1]
```diff
@@ -15,15 +15,17 @@ Demo
...
-*   [Highmaps demo](https://highcharts.com/maps/demo/)
+*   [Highcharts Stock demo](https://highcharts.com/stock/demo/)
+*   [Highcharts Maps demo](https://highcharts.com/maps/demo/)
+*   [Highcharts Gantt demo](https://highcharts.com/gantt/demo/)
 
 API
 ---
 
-For more specific information on Highcharts and Highstock options and functions, visit our API sites which also include several live and customizeable examples.
+For more specific information on Highcharts and Highcharts Stock options and functions, visit our API sites which also include several live and customizeable examples.
```
Comment (TorsteinHonsi) [inline docs/Readme.md]: We can rephrase this to "... information on Highcharts options and functions ..."

Since the Highcharts name now includes Stock, Maps and Gantt, plus the main packages are listed below.

File: docs/getting-started/install-from-npm.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 Installing with NPM
 ===
 
-The official npm package contains Highcharts, Highstock and Highmaps plus all modules. Start by installing Highcharts as a node module and save it as a dependency in your package.json:
+The official npm package contains Highcharts, Highcharts Stock and Highcharts Maps plus all modules. Start by installing Highcharts as a node module and save it as a dependency in your package.json:
```
Comment (TorsteinHonsi) [inline docs/getting-started/install-from-npm.md]: Rephrase to "...contains Highcharts, including the Stock, Maps and Gantt packages, plus all modules"

File: docs/getting-started/system-requirements.md [hunk 3]
```diff
@@ -3,9 +3,9 @@ System requirements
 
 Highcharts is solely based on native browser technologies and doesn't require client side plugins like Flash or Java. Furthermore you don't need to install anything on your server. No PHP or ASP.NET. Highcharts needs only the highcharts.js core to run.
 
-Highcharts and Highstock work in all modern browsers including mobile devices and Internet Explorer from version 6. Standard browsers use [SVG](https://www.w3.org/Graphics/SVG/) for the graphics rendering. In legacy Internet Explorer (IE8 and before) graphics are drawn using [VML](https://www.w3.org/TR/NOTE-VML).
+Highcharts and Highcharts Stock work in all modern browsers including mobile devices and Internet Explorer from version 6. Standard browsers use [SVG](https://www.w3.org/Graphics/SVG/) for the graphics rendering. In legacy Internet Explorer (IE8 and before) graphics are drawn using [VML](https://www.w3.org/TR/NOTE-VML).
```
Comment (TorsteinHonsi) [inline docs/getting-started/system-requirements.md]: Just mention Highcharts, it includes Stock, Maps and Gantt.

File: docs/getting-started/system-requirements.md [hunk 4]
```diff
@@ -3,9 +3,9 @@ System requirements
...
 
-Highcharts and Highstock run on any server that supports HTML. You can even run Highcharts locally from a filesystem, since all the rendering is done locally in a browser.
+Highcharts and Highcharts Stock run on any server that supports HTML. You can even run Highcharts locally from a filesystem, since all the rendering is done locally in a browser.
```
Comment (TorsteinHonsi) [inline docs/getting-started/system-requirements.md]: Just mention Highcharts, it includes Stock, Maps and Gantt.

File: docs/public-index.md [hunk 5]
```diff
@@ -13,15 +13,16 @@ Demo
...
-*   [Highmaps demo](https://highcharts.com/maps/demo/)
+*   [Highcharts Stock demo](https://highcharts.com/stock/demo/)
+*   [Highcharts Maps demo](https://highcharts.com/maps/demo/)
+*   [Highcharts Gantt demo](https://highcharts.com/gantt/demo/)
 
 API
 ---
 
-For more specific information on Highcharts and Highstock options and functions, visit our API sites which also include several live and customizeable examples.
+For more specific information on Highcharts and Highcharts Stock options and functions, visit our API sites which also include several live and customizeable examples.
```
Comment (TorsteinHonsi) [inline docs/public-index.md]: See the first comment

File: docs/getting-started/system-requirements.md [hunk 6]
```diff
@@ -3,9 +3,9 @@ System requirements
 
 Highcharts is solely based on native browser technologies and doesn't require client side plugins like Flash or Java. Furthermore you don't need to install anything on your server. No PHP or ASP.NET. Highcharts needs only the highcharts.js core to run.
 
-Highcharts and Highcharts Stock work in all modern browsers including mobile devices and Internet Explorer from version 6. Standard browsers use [SVG](https://www.w3.org/Graphics/SVG/) for the graphics rendering. In legacy Internet Explorer (IE8 and before) graphics are drawn using [VML](https://www.w3.org/TR/NOTE-VML).
+Highcharts work in all modern browsers including mobile devices and Internet Explorer from version 6. Standard browsers use [SVG](https://www.w3.org/Graphics/SVG/) for the graphics rendering. In legacy Internet Explorer (IE8 and before) graphics are drawn using [VML](https://www.w3.org/TR/NOTE-VML).
```
Comment (TorsteinHonsi) [inline docs/getting-started/system-requirements.md:6]: Highcharts _works_ ...

File: docs/getting-started/system-requirements.md [hunk 7]
```diff
@@ -3,9 +3,9 @@ System requirements
...
 
-Highcharts and Highcharts Stock run on any server that supports HTML. You can even run Highcharts locally from a filesystem, since all the rendering is done locally in a browser.
+Highcharts run on any server that supports HTML. You can even run Highcharts locally from a filesystem, since all the rendering is done locally in a browser.
```
Comment (TorsteinHonsi) [inline docs/getting-started/system-requirements.md:8]: Highcharts _runs_

Comment (goransle) [reply docs/getting-started/system-requirements.md]: Now it does 🏃
